core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts

```diff
@@ -16,219 +16,219 @@
-              userContributionRights.can_suggest_questions);
-            // TODO(#8521): Remove the use of $rootScope.$apply()
-            // once the controller is migrated to angular.
-            $rootScope.$applyAsync();
-          });
-
-        UserService.getUserInfoAsync().then(function(userInfo) {
-          ctrl.userInfoIsLoading = false;
-          if (userInfo.isLoggedIn()) {
-            ctrl.userIsLoggedIn = true;
-            ctrl.username = userInfo.getUsername();
-          } else {
-            ctrl.userIsLoggedIn = false;
-            ctrl.username = '';
-          }
-          // TODO(#8521): Remove the use of $rootScope.$apply()
-          // once the controller is migrated to angular.
-          $rootScope.$applyAsync();
-        });
-
-        ctrl.topicName = DEFAULT_OPPORTUNITY_TOPIC_NAME;
-        TranslationTopicService.setActiveTopicName(ctrl.topicName);
-
-        ContributionOpportunitiesService.getTranslatableTopicNamesAsync()
-          .then(function(topicNames) {
-            if (topicNames.indexOf(prevSelectedTopicName) !== -1) {
-              ctrl.topicName = prevSelectedTopicName;
-              TranslationTopicService.setActiveTopicName(ctrl.topicName);
-            }
-            $rootScope.$applyAsync();
-          });
-
-        ctrl.activeTabName = 'myContributionTab';
-        ctrl.tabsDetails = CONTRIBUTOR_DASHBOARD_TABS_DETAILS;
-        ctrl.OPPIA_AVATAR_LINK_URL = OPPIA_AVATAR_LINK_URL;
-        ctrl.OPPIA_AVATAR_IMAGE_URL = (
-          UrlInterpolationService.getStaticImageUrl(
-            '/avatar/oppia_avatar_100px.svg'));
-      };
+import { Component, OnInit } from '@angular/core';
+import { SafeUrl } from '@angular/platform-browser';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import AppConstants from 'assets/constants';
+import { ContributorDashboardConstants } from 'pages/contributor-dashboard-page/contributor-dashboard-page.constants';
+import { ContributionAndReviewService } from './services/contribution-and-review.service';
+import { ContributionOpportunitiesService } from './services/contribution-opportunities.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { LocalStorageService } from 'services/local-storage.service';
+import { SvgSanitizerService } from 'services/svg-sanitizer.service';
+import { TranslationLanguageService } from 'pages/exploration-editor-page/translation-tab/services/translation-language.service';
+import { TranslationTopicService } from 'pages/exploration-editor-page/translation-tab/services/translation-topic.service';
+import { UserService } from 'services/user.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface TabsDetails {
+  submitQuestionTab: {
+    enabled: boolean;
+  };
+}
```

Comment:
Is this type necessary if it's just being assigned the constant defined in core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.constants.ts?

Reply:
on line no. 185 it is throwing type error. `any Property 'submitQuestionTab' does not exist` + we are using this variable in there different functions

Reply:
hm okay then shouldn't we create a type for the constant in core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.constants.ts and import here instead?

Reply:
done